bson/__init__.py

```diff
@@ -896,12 +896,20 @@ def _name_value_to_bson(
     in_fallback_call: bool = False,
 ) -> bytes:
     """Encode a single name, value pair."""
+
+    was_integer_overflow = False
+
     # First see if the type is already cached. KeyError will only ever
     # happen once per subtype.
     try:
         return _ENCODERS[type(value)](name, value, check_keys, opts)  # type: ignore
     except KeyError:
         pass
+    except OverflowError as error:
+        if error.args != ("BSON can only handle up to 8-byte ints",):
```

Comment:
Could you replace this with `if not isinstance(value, int):`? This way the fallback encoder won't be called multiple times for one OverflowError in this case:
```
>>> bson.encode({'a' {'b': 'c': 2<<65}}, codec_options=CodecOptions(...fallback_encoder=lambda val: val)))

```